.github/workflows/robot/internal/bot/backport.go

```diff
@@ -50,6 +50,16 @@ func (b *Bot) Backport(ctx context.Context) error {
 		return trace.Wrap(err)
 	}
 
+	// If this workflow is running on a release branch, then it means
+	// a backport was merged and there's no need to open additional
+	// backport PRs. We can safely delete the remote branch though,
+	// because we know this is an internal contributor's merged PR.
+	if isReleaseBranch(b.c.Environment.UnsafeBase) &&
+		isBotBackportBranch(b.c.Environment.UnsafeHead) &&
+		!pull.Fork {
+		return trace.Wrap(git("push", "origin", "--delete", b.c.Environment.UnsafeHead))
+	}
```

Comment:
Will this work? If this runs on the backport branch it will delete the backport branch before it gets merged?

Reply:
This workflow runs when _any_ PR is merged.

When a PR is merged to master, it opens a backport.

When a PR is merged to a release branch (meaning a backport PR was merged) it used to be a no-op. We're changing the behavior here from "do nothing and exit" to "delete the branch that was just merged."